### DISCUSSION TITLE
Feature idea: random keystroke delay

### DISCUSSION THREAD

**Discussion by soupslurpr** [florisboard/florisboard#2165] [General]
Hi! I have an idea for a feature and that is, random keystroke delay, so it would be a random delay between when you press a key and it actually getting typed. This feature would be to prevent fingerprinting based on keystrokes. I have made a fork that adds this in here https://github.com/SoupSlurpr/florisboard, but it probably isn't optimal as I have basically no android development experience. There are some issues with it such as crashing if it gets too far behind, but otherwise it does seem to at least do its function properly. You can try this website and press the info button to see a percentage. https://www.typingdna.com/demo-sametext.html

**Comment by codenyte** [comment]
So this would kinda be similar to "kloak" on Whonix/Linux?

**Reply by soupslurpr** [reply]
Yeah.

**Reply by codenyte** [reply]
Nice, hope this gets implemented

**Reply by soupslurpr** [reply]
The code I wrote is not something that should be implemented as it is, because it's low quality since it freezes the keyboard UI and also crashes when it gets too far behind. I am very new to android development, and back when I made that I basically just took the shortest route that I could see at the time to have that feature, and I had very little to no experience. Perhaps in the future I could actually attempt at doing it correctly so that those two issues are fixed, but I'm not sure if I'm interested.

**Reply by codenyte** [reply]
Didn't realize you actually implemented it yourself, I thought this was more like a feature request to the devs

**Reply by soupslurpr** [reply]
Ah, yes it is a feature request to the devs, I just also tried it myself lol
